src/config-management.ts

```diff
@@ -171,23 +176,38 @@ function _isAvailablePeriodValid(availablePeriod: AvailablePeriod, index: number
 
 /**
  * Indicate either if the provided date string is valid or not.
- * @param {string} dateString The date string to check.
+ * @param {PeriodMoment} periodMoment The date string to check.
  * @returns {boolean}
  */
-function _isDateStringValid(dateString: string) {
-	const longFormat = "YYYY-MM-DD HH:mm"
-	const shortFormat = "MM-DD HH:mm"
-	/* Format YYYY-MM-DD HH:mm */
-	if (dateString.length === 16) {
-		return dayjs(dateString, longFormat, true).format(longFormat) === dateString
+function _isPeriodMomentValid(periodMoment: PeriodMoment) {
+	if (periodMoment.hour == null && periodMoment.minute != null) {
+		return false
 	}
 
-	/* Format MM-DD HH:mm */
-	if (dateString.length === 11) {
-		return dayjs(dateString, shortFormat).format(shortFormat) === dateString
+	const isYearAndMonthValid = (
+		(periodMoment.year == null || periodMoment.year > 0)
+		&& periodMoment.month >= 0 && periodMoment.month <= 11
+	)
+
+	if (!isYearAndMonthValid) {
+		return false
 	}
 
-	return false
+	/* The day check depends on month and year */
+	let day = dayjs().month(periodMoment.month)
+	if (periodMoment.year) { day = day.year(periodMoment.year) }
+	day = day.date(periodMoment.day)
+
+	const isDayValid = (
+		periodMoment.day >= 1 && periodMoment.day <= 31
```

Comment:
Little bit of a nitpicking but it the year is given we could use : https://day.js.org/docs/en/display/days-in-month for the upper limit.

Reply:
Oh yes, good tip, thanks!